packages/main/cypress/specs/Popover.cy.tsx

```diff
@@ -49,254 +147,919 @@ describe("Accessibility", () => {
+
+		cy.get("[ui5-button]").then($btnClosePopover => {
+			$btnClosePopover.get(0).addEventListener("click", () => {
+				cy.get("[ui5-popover]").invoke("prop", "open", false);
+			});
+		});
+		cy.get("[ui5-button]").realClick();
+	});
+
+	it("tests 'opener' set as DOM ref of abstract element's DOM reference", () => {
+		cy.mount(
+			<>
+				<Toolbar id="tb">
+					<ToolbarButton text="Add"></ToolbarButton>
+					<ToolbarButton text="Delete"></ToolbarButton>
+					<ToolbarButton id="btnOpenPopover" text="Open Menu" preventOverflowClosing={true}></ToolbarButton>
+				</Toolbar>
+				<Popover id="popup">
+					<Button id="btnClosePopover">Close</Button>
+				</Popover>
+			</>
+		);
+
+		cy.get("#btnOpenPopover").then($toolbarBtn => {
+			cy.wrap($toolbarBtn.get(0)).as("toolbarBtn");
+		});
+
+		cy.get("@toolbarBtn").then($toolbarBtn => {
+			cy.get("[ui5-popover]").invoke("prop", "opener", $toolbarBtn.get(0));
+		});
+		cy.get("[ui5-popover]").invoke("prop", "open", true);
+
+		cy.get("[ui5-button]").then($btnClosePopover => {
+			$btnClosePopover.get(0).addEventListener("click", () => {
+				cy.get("[ui5-popover]").invoke("prop", "open", false);
+			});
+		});
+	});
+
+	it("tests calling _showOutsideViewport method", () => {
+		cy.mount(
+			<>
+				<Button id="btnOpen">Open</Button>
+				<Popover id="popover" opener="btnOpen">
+					<Button id="btnClosePopover">Close</Button>
+				</Popover>
+			</>
+		);
+
+		cy.get("[ui5-popover]").then(el => {
+			cy.spy<Popover>((el.get(0) as Popover), "_showOutsideViewport").as("showOutsideViewport");
+		});
+
+		cy.get("[ui5-popover]").invoke("prop", "open", "true");
+
+		cy.get("@showOutsideViewport")
+			.should("have.been.calledOnce");
+	});
+
+	it("tests calling _showOutsideViewport method, when popover is created dynamically", () => {
+		cy.mount(
+			<>
+				<div id="container"></div>
+				<Button id="btnOpen">Open</Button>
+			</>
+		);
+
+		cy.get("#container").then(container => {
+			const popover = document.createElement("ui5-popover");
+
+			cy.spy<Popover>((popover as Popover), "_showOutsideViewport").as("showOutsideViewport");
+
+			popover.id = "popover";
+			popover.headerText = "Popover Header";
+			popover.opener = "btnOpen";
+			popover.open = true;
+
+			const content = document.createElement("div");
+			content.innerHTML = "<button id='popoverBtn'>button</button>";
+			popover.appendChild(content);
+
+			container.get(0).appendChild(popover);
+		});
+
+		cy.get<Popover>("[ui5-popover]")
+			.ui5PopoverOpened();
+
+		cy.get("@showOutsideViewport")
+			.should("have.been.calledOnce");
+	});
+
+	it("tests popover toggling", () => {
+		cy.mount(
+			<>
+				<Button id="btn" />
+				<Popover opener="btn" />
+				<br />
+				<Button id="btn2"/>
+			</>);
+
+		cy.get("[ui5-popover]").invoke("prop", "open", "true");
+
+		cy.get<Popover>("[ui5-popover]").ui5PopoverOpened();
+
+		cy.get("#btn2").realClick();
+		cy.get("[ui5-popover]").should("not.have.attr", "open");
```

Comment:
Clicking this button is obsolete. Rather "invoke prop open false" and then check the dom reference for "is displayed"